File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/eq-master/lib/services/fitness_service.dart
import '../models/api_models.dart';
import 'api_client.dart';

class FitnessService {
  final ApiClient _api = ApiClient.instance;

  Future<List<FitnessRecordDto>> getPlayerFitness(
      String clubId, String teamId, String playerId) async {
    final json =
        await _api.get('/clubs/$clubId/teams/$teamId/players/$playerId/fitness');
    return (json as List)
        .map((e) =>
            FitnessRecordDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<FitnessRecordDto> createFitnessRecord(
      String clubId, String teamId, String playerId, Map<String, dynamic> body) async {
    final json = await _api.post(
      '/clubs/$clubId/teams/$teamId/players/$playerId/fitness',
      body: body,
    );
    return FitnessRecordDto.fromJson(Map<String, dynamic>.from(json as Map));
  }
}

File: react_web/src/services/fitnessService.js
import api from './api';

// POST /clubs/:clubId/teams/:teamId/players/:playerUserId/fitness
export const createFitnessRecord = async (clubId, teamId, playerUserId, request) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/fitness`, request);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/players/:playerUserId/fitness
export const getPlayerFitnessRecords = async (clubId, teamId, playerUserId) => {
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/fitness`);
  return data;
};

// GET /players/me/fitness
export const getMyFitnessRecords = async () => {
  const { data } = await api.get('/players/me/fitness');
  return data;
};
